web/src/channel.ts

```diff
@@ -22,6 +22,17 @@ type AjaxRequestHandlerOptions = Omit<JQuery.AjaxSettings, "success"> & {
           ) => void)
         | undefined;
     error?: JQuery.Ajax.ErrorCallback<unknown>;
+    idempotencyKeyManager?: IdempotencyKeyManager;
+};
+
+// A generic idempotency manager for any future request we want to make it idempotent.
+// Passed as HTTP header (if defined) to $.ajax().
+// success and error are not used yet but are kept in case
+// we decide to handle each case separately.
```

Comment:
If success and error are not used yet, then I would just omit them from this PR.  You can still comment future directions, but I wouldn't introduce dead code here.